### ISSUE TITLE
Background application does not appear in binding interface in SteamVR

### ISSUE THREAD

**Issue by BOLL7708** [ValveSoftware/openvr#879]
Hello! I'm working on my C# OpenVR to JSON application, it simply outputs data from OpenVR upon web requests to user in streaming overlays. I've been using it successfully for almost two years already.

What I'm trying to do now is convert it to the new input system and I'm working my way through the documentation but have gotten stuck at the point where I'm supposed to build a binding file using the controller binding interface.

At first I misspelled the path to the action manifest, that popped up in the headset as an error message when going to the controller bindings in the settings, so clearly my application was detected but it wasn't shown. I had to restart SteamVR to have it use the correct manifest path so the error message went away, but after that my application still won't show up.

My other calls to OpenVR are working and I'm getting empty data out of an action, it has no binding right now due to me not being able to use to tool so clearly nothing will be grabbed from any of the controllers, but the calls works.

**TL;DR:** My question is, will background applications currently pop up in the controller binding interface at all? Or, should I look closer for issues in my code? 😅

I'm using the latest C# headers and the 64bit Windows .dll as of today (Sep 7th).

**Comment by BOLL7708** [comment]
Here's a screenshot if I empty the binding_url in the manifest. It was pointing to an empty file which gave no message, but this pops up if I clear the field.
![image](https://user-images.githubusercontent.com/12276837/45242379-10e2e300-b2f0-11e8-890e-1ebc538262aa.png) Which makes me think the system recognizes the application, but again... it never shows up in the list to edit bindings for 🤔 The manifest I'm using is the sample one from [here](https://github.com/ValveSoftware/openvr/wiki/Action-manifest), with the binding_url changed and an extra action added.

**Comment by BOLL7708** [comment]
I experimented today with using a non-steam application, but one that actually renders to VR, my application never does that. This app, ChaperoneTweak, popped up in the controller binding interface and I could export the default controller bindings for me to experiment with.

While matching the actions between the manifest and bindings in my app I still get no input. I suspect this is because I have no idea what `app_key` I am supposed to use. In the export there is an app key for an external steam application: _2147483651_

I tried adding my own application to Steam, it would still not show up in the bindings interface, but I took a screenshot that ends up in a folder with the number _9564780782543699968_, which is... way longer than the prior so I'm suspecting it's not the same type of ID.

Regardless, tried running my application with either ID as app key, no dice. Switched around which input I tried to output... still nothing. I'll continue digging I guess, but right now it feels like I have to acquire a `steam.app.###` key somehow.

**Comment by BOLL7708** [comment]
I got some help over at slack, and now I have an app.vrmanifest to specify my app key, I keep getting `VREvent_Input_BindingLoadFailed` though, and my app still does not show up in the binding interface so I can make an actual bindings file instead of trying to write it myself 🤔 

I guess overlay and background applications just won't show up, this as Advanced OpenVR Settings and OpenVR Input Emulator won't show up even if they have overlays. For overlays it makes sense I guess, as they are interacted with in a standard manner, with my background application I want to listen to everything though.

**Comment by BOLL7708** [comment]
The vrserver.txt log file was pointed out to me, so now I'm monitoring it, but it doesn't really give me any more information on what is going wrong.
`App boll7708.openvr2json (vive_controller) failed to load binding file. Input will not be available`
I get this regardless if I specify a default binding section in my action manifest or not. And trying to get a generated binding is still fruitless as my background application will not appear in the binding interface, in headset or on webpage, so #9 in [this guide](https://github.com/ValveSoftware/openvr/wiki/SteamVR-Input) is impassable. 🤔

**Comment by BOLL7708** [comment]
Did a test project [here](https://github.com/BOLL7708/OpenVRInputTest), bindings still won't load and I'm fairly clueless still 😕 If anyone cares to take a look, be my guest. 

It's a C# project built in Visual Studio 2017, build for 64x. If settings don't carry over (not sure at this point) make sure to set "always copy" in the properties for the openvr_api files, vrmanifest and json files.

**Comment by username223** [comment]
It appears that SteamVR beta 1.3.1 has fixed this.

My background application now shows up at the bottom of the list without initializing as Scene or changing the `.vrappconfig` files `last_launch_time`.

**Comment by BOLL7708** [comment]
@username223 whoa, look at that, I now see my accessory apps in the list. 

I have no idea if this is the only thing needed to finally get to reading input again, but at least it will help with completing the input integration guide. Nice that you saw this, I read the update notes and saw updates to input but figured this was not fixed. I'll have to finish up my current project so I can get back to integrating input for my original app 😅

I'll close this, if I still have issues with the interface I'll reopen or more likely open a new ticket.
